### ISSUE TITLE
Timeout during sync resulting in sync not closing

### ISSUE THREAD

**Issue by nikparo** [Backblaze/B2_Command_Line_Tool#206]
While testing the sync functionality, I decided to turn my VPN off mid sync. After a good while (at least 10min?) uploads managed to continue. A while later, I turned on VPN again, and uploads again stopped for quite a while. However, a while later I got two errors:

```
b2_upload(/mnt/data/borgtest/data/0/249, data/0/249, 1470290017627): UnknownError() unknown error: ReadTimeout(ReadTimeoutError("HTTPSConnectionPool(host='pod-000-1005-09.backblaze.com', port=443): Read timed out. (read timeout=None)",),)
b2_upload(/mnt/data/borgtest/data/0/250, data/0/250, 1470290019022): UnknownError() unknown error: ReadTimeout(ReadTimeoutError("HTTPSConnectionPool(host='pod-000-1009-00.backblaze.com', port=443): Read timed out. (read timeout=None)",),)
```
Those two files may have been uploading when I turned VPN on, but I imagine others would have been as well.

Anyhow, after the sync finished uploading all other files, the sync progress remained stuck at 289/293, (this seems to be a bug as well, I'm pretty sure 291 files managed to upload fine) and did nothing for at least 40min after which I killed it.

Running `b2 sync` again it happily uploaded the remaining two(!) files.

TL:DR
- Upload timeouts very slowly (may be ok) but doesn't seem to retry if it fails
- Progress status may report faulty uploads when there have been Errors.
- Sync remained open, waiting (forever?) for two failed uploads to complete.

**Comment by ppolewicz** [comment]
http://docs.python-requests.org/en/master/api/#exceptions
